src/IO/Observer/VolumeActions.hpp

```diff
@@ -231,15 +233,18 @@ struct WriteVolumeData {
       h5::H5File<h5::AccessType::ReadWrite> h5file(
           file_prefix + std::to_string(Parallel::my_node()) + ".h5", true);
       constexpr size_t version_number = 0;
-      auto& volume_file =
-          h5file.try_insert<h5::VolumeData>(subfile_name, version_number);
-      for (const auto& id_and_tensor_data_for_grid : volume_data) {
-        const auto& extents_and_tensors = id_and_tensor_data_for_grid.second;
-        volume_file.insert_tensor_data(
-            observation_id.hash(), observation_id.value(), extents_and_tensors);
+      auto& volume_file = h5file.try_insert<h5::VolumeData>(
+          subfile_name, version_number);
+      std::vector<ExtentsAndTensorVolumeData> dg_elements;
+      dg_elements.reserve(volume_data.size());
+      for (auto id_and_element : volume_data) {
+        dg_elements.push_back(id_and_element.second);
       }
+      // Write the data to the file
+      volume_file.write_volume_data(observation_id.hash(),
+                                    observation_id.value(), dg_elements);
+      Parallel::unlock(&file_lock);
```

Comment:
The `unlock` needs to be outside of the braces. This won't work in parallel as is